core/spam/engine.go

```diff
@@ -118,6 +124,18 @@ func New(log *logging.Logger, config Config, epochEngine EpochEngine, accounting
 	return e
 }
 
+// OnEpochDurationChanged updates the ban duration as a fraction of the epoch duration.
+func (e *Engine) OnEpochDurationChanged(_ context.Context, duration time.Duration) error {
+	epochImpliedDurationNano, _ := num.UintFromDecimal(num.DecimalFromInt64(duration.Nanoseconds()).Mul(banDurationAsEpochFraction))
+	epochImpliedDurationDuration := time.Duration(epochImpliedDurationNano.Uint64())
```

Comment:
Do we need to go through a num.UintFromDecimal here? decimal.IntPart would be enough? doesn't matter much thought

Reply:
for the frequency this happens it doesn't matter really